kernel/src/process.rs

```diff
@@ -1465,9 +1465,16 @@ impl<C: Chip> ProcessType for Process<'_, C> {
         }
     }
 
-    unsafe fn set_byte(&self, addr: *mut u8, value: u8) -> bool {
+    // This function verifies that the dereference is valid and safe.
+    #[allow(clippy::not_unsafe_ptr_arg_deref)]
+    fn set_byte(&self, addr: *mut u8, value: u8) -> bool {
         if self.in_app_owned_memory(addr, 1) {
-            *addr = value;
+            // This is safe because we have verified that the address we are
+            // writing to is in fact within the process's process-accessible
+            // memory.
```

Comment:
is this check really sufficient for memory safety? I think it is only sufficient if we can guarantee that no other reference to this byte exists anywhere in the kernel. Which I think currently is not true (https://github.com/tock/tock/issues/2369).

Reply:
On cortex-m every function in this PR writes to process memory eventually. We can take care of #2369 pretty quickly, right?

Reply:
Yeah I think we can take care of #2369 pretty quickly. But I think we should clearly document that this function is only safe if no references to the byte being modified exist anywhere in the kernel. For example it might still be unsafe to call this function early in the board setup process when we still have references to process memory.

Reply:
If `set_byte` cannot verify that this is safe, and there are no "tock level" restrictions (convention or otherwise) that would guarantee this is safe, then I think we should leave this function marked `unsafe` because the caller then needs to do additional checks.

However, I would rather have rules around when a reference to process memory can exist than have to consider the implications of what if the reference does exist for every function in process and every use of those functions.

Reply:
Yes, I think having rules around when a reference to process memory can exist would be a good solution here.